### PR TITLE
Add Debian package framework and "make install" target

### PR DESCRIPTION
**Pull request by ptudor**

### CODE REVIEW

**Comment by akhepcat** [comment]
Just for future-proofing against installed conflicts,  debian/control  really needs to have a 
"Conflicts: gpsd-clients"    line, as per https://www.debian.org/doc/debian-policy/ch-relationships.html#conflicting-binary-packages-conflicts
so we don't unintentionally overwrite an installed version of ubxtool provided by the gps packages.

* ubuntu/focal  is in testing right now, so this is currently an issue starting at this release
* debian/bullseye is not yet an issue, but should be "pre-tracked" 

This line can be removed if/when @ahupowerdns renames "ubxtool"  to something non-conflicting

File: Makefile [hunk 1]
```diff
@@ -33,6 +40,30 @@ clean:
...
+	rm -f $(DESTDIR)$(prefix)/share/man/man1/testrunner.1.gz
+
+install: $(PROGRAMS) help2man
+	$(INSTALL) -m 755 -d $(DESTDIR)$(prefix)/bin
+	$(foreach binaryfile,$(PROGRAMS),$(INSTALL) -s -m 755 -D ./$(binaryfile) $(DESTDIR)$(prefix)/bin/$(binaryfile);)
+	@echo "using cp instead of install because recursive directories of ascii"
+	mkdir -p $(DESTDIR)$(prefix)$(htdocs)/galmon
+	cp -a html $(DESTDIR)$(prefix)$(htdocs)/galmon/
+
+install-debian:
```
Comment (ahupowerdns) [inline Makefile]: This confuses me greatly. I'd assume 'make install-debian' would mean "install what you just compiled in reasonable Debian compatible paths" and not "reconfigure apt to get packages from ota.bike".

File: Makefile [hunk 2]
```diff
@@ -33,6 +40,30 @@ clean:
...
+	apt-key adv --fetch-keys https://ota.bike/public-package-signing-keys/86E7F51C04FBAAB0.asc
+	echo "deb https://ota.bike/debian/ buster main" > /etc/apt/sources.list.d/galmon.list
+	apt-get update && apt-get install -y galmon
+
+install-raspbian:
```
Comment (ahupowerdns) [inline Makefile]: Same here of course. It is good to have this option, but 'install-' is not the name for it.

File: PACKAGE-DEBIAN.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+better suited for smooth building. For comparison, a VM on a low-end AMD Ryzen 3 2200G builds the package at -j1 in about two minutes.
+
+These are fast multi-core computers but we turn off parallel compiles because of limited RAM.
+Limiting optimizations to -O0 cuts the compile time in half approximately.
+
+### Why do this?
+
+Convenience, uniformity, and scalability:
+Hand-compiling software is fun, but vendor package management solutions
+exist to give us reliable unattended installations for free.
```
Comment (ahupowerdns) [inline PACKAGE-DEBIAN.md]: +1 - thanks!

File: debian/control [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+Package: galmon
+Architecture: any
+Multi-Arch: foreign
+Conflicts: gpsd-clients
+Depends: ${misc:Depends}, ${shlibs:Depends}
+Description: galmon GNSS Monitoring Project software
+                         88                                             
+                         88                                             
+                         88                                             
+   ,adPPYb,d8 ,adPPYYba, 88 88,dPYba,,adPYba,   ,adPPYba,  8b,dPPYba,
```
Comment (ahupowerdns) [inline debian/control]: aww yess

**Comment by ptudor** [comment]
Because there's not a direct reply box for it, a reply here for the question about automatic startups and interface binds at installation:

Installing the packaging only updates the bits at rest. Positive user interaction is required to enable or run any installed binaries. Because that is the most likely intent, after package installation completes a note is printed to the console with copy-and-paste text to enable the daemon.

Updating the package does go a step further. If the daemon is already running, updating an already installed package will restart an already enabled daemon. Nothing happens where items remain in their default disabled state.

Interface choices are left to the administrator but the installed default examples demonstrate IPv6 localhost.

File: Makefile (hunk 1 above)
Comment (ptudor) [reply Makefile]: Trying "download-debian-package" as a distinct placeholder.

File: Makefile (hunk 2 above)
Comment (ptudor) [reply Makefile]: Trying "download-raspbian-package" as a distinct placeholder.

**Pull request merged**